api/integrations/flagsmith/client.py

```diff
@@ -21,14 +19,24 @@
 from integrations.flagsmith.exceptions import FlagsmithIntegrationError
 from integrations.flagsmith.flagsmith_service import ENVIRONMENT_JSON_PATH
 
-_flagsmith_client: typing.Optional[Flagsmith] = None
+_flagsmith_clients: dict[str, Flagsmith] = {}
 
 
-def get_client() -> Flagsmith:
-    global _flagsmith_client
+def get_client(name: str = "default", local_eval: bool = False) -> Flagsmith:
+    global _flagsmith_clients
 
-    if not _flagsmith_client:
-        _flagsmith_client = Flagsmith(**_get_client_kwargs())
+    try:
+        _flagsmith_client = _flagsmith_clients[name]
+    except (KeyError, TypeError):
+        kwargs = _get_client_kwargs()
+        kwargs["enable_local_evaluation"] = local_eval
+        _flagsmith_client = Flagsmith(**kwargs)
+        # TODO: Remove this once client has been fixed.
```

Comment:
Let's link the relevant PR.

Reply:
Done, although I'm sorta hoping we get that PR merged and deployed before this code is ready to merge.

Reply:
I've just released v3.6.0 with this in it.

Reply:
Ok great I've removed the temporary shim.

Reply:
Well, I'm expecting to also see an updated poetry.lock to update the flagsmith client lib in this PR, right?

Reply:
Great catch! I've added the update which came with a bunch of dependency upgrades, which pulled in an update to the `flagsmith-flag-engine` library which required a series of updates to a bunch of tests and some changes to the code that was using the `flagsmith-flag-engine` library.